object-filter/src/main/java/org/infinispan/objectfilter/impl/hql/ProtobufPropertyHelper.java

```diff
@@ -126,6 +134,32 @@ public boolean hasEmbeddedProperty(String entityType, List<String> propertyPath)
    }
 
    @Override
+   public boolean isRepeatedProperty(String entityType, List<String> propertyPath) {
+      Descriptor messageDescriptor;
+      try {
+         messageDescriptor = serializationContext.getMessageDescriptor(entityType);
+      } catch (Exception e) {
+         throw new IllegalStateException("Unknown entity name " + entityType);
+      }
+
+      int i = 0;
+      for (String p : propertyPath) {
+         FieldDescriptor field = messageDescriptor.findFieldByName(p);
+         if (field == null) {
+            break;
+         }
+         if (field.isRepeated()) {
+            return true;
+         }
+         if (++i == propertyPath.size() || field.getJavaType() != JavaType.MESSAGE) {
```

Comment:
Do you really need the `++i == propertyPath.size()` condition here, if you're already iterating on the propertyPath?